GitUI/Editor/FileViewer.cs

```diff
@@ -857,26 +899,26 @@ private void ResetForText([CanBeNull] string fileName)
             {
                 internalFileViewer.SetHighlightingForFile(fileName);
             }
-
-            if (!string.IsNullOrEmpty(fileName) &&
-                (fileName.EndsWith(".diff", StringComparison.OrdinalIgnoreCase) ||
-                 fileName.EndsWith(".patch", StringComparison.OrdinalIgnoreCase)))
-            {
-                ResetForDiff();
-            }
         }
 
-        private void ResetForDiff()
+        private void ResetForDiff([CanBeNull] string fileName)
         {
-            Reset(true, true);
-            internalFileViewer.SetHighlighting("");
-            _patchHighlighting = true;
+            Reset(true, true, fileName);
+
+            if (ShowSyntaxHighlightingInDiff && fileName != null)
+            {
+                internalFileViewer.SetHighlightingForFile(fileName);
+            }
+            else
+            {
+                internalFileViewer.SetHighlighting("");
+            }
         }
 
-        private void Reset(bool diff, bool text, bool isStagingDiff = false)
+        private void Reset(bool diff, bool text, [CanBeNull] string fileName)
```

Comment:
Can we rename the args to communicate the intent better? E.g.:
```suggestion
        private void Reset(bool isDiff, bool isText, [CanBeNull] string fileName)
```

Reply:
Done